hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockInputStream.java

```diff
@@ -567,6 +673,10 @@ synchronized long getBlockPosition() {
     return blockPosition;
   }
 
+  public synchronized FileInputStream getBlockInputStream() {
```

Comment:
This method doesn't need to be synchronized. Just declare blockInputStream volatile.